### DISCUSSION TITLE
what does redis do ?

### DISCUSSION THREAD

**Discussion by FarisZR** [searxng/searxng#993] [Q&A] [answered]
Hi

I have just noticed that in the new settings.yml file there is a redis section,
It's not very clear from the documentation what it does.
Does it cache results for reuse ? What is the cache invalidation policy ? 
Or does it do something else ?

**Comment by mrpaulblack** [comment, marked as answer]
It is for the new limiter plugin that replaces filtron for now. IP addresses are getting stored with a hash so the limiter plugin knows who to block. Also in the future we will use redis for different caches more (not results or images since this would be a privacy invasion, but other stuff for sure).

So basically SearXNG switched from SearXNG + filtron + morty to SearXNG core + redis :+1:

**Reply by FarisZR** [reply]
Great!, thanks for your answer. It would be great if you could include this in the documentation.

**Reply by return42** [reply]
I send PR https://github.com/searxng/searxng/pull/994

**Comment by flori** [comment]
I still couldn't find a reason for why anyone would want set up redis for searxng in the documentation (apparently not to cache search results). Ironically I had to search with google to find this answer.

**Reply by return42** [reply]
> Ironically I had to search with google to find this answer.

https://docs.searxng.org/admin/searx.limiter.html#limiter .. and some of engines are using redis to store tokens.

**Comment by flori** [comment]
As I said I have found the answer with google. The next person coming along probably will still be puzzled by

> Redis DB Implementation of the redis client This implementation uses the redis setup from settings.yml. A redis DB connect can be tested by…

after clicking on "Redis DB" and maybe wonder if he can somehow search the redis db with SearXNG.

**Reply by return42** [reply]
@flori If you think the docs can be improved: your PRs are welcome.
